src/wazuh_modules/agent_upgrade/wm_agent_upgrade_tasks.c

```diff
@@ -0,0 +1,78 @@
+/*
+ * Wazuh Module for Agent Upgrading
+ * Copyright (C) 2015-2020, Wazuh Inc.
+ * July 3, 2020.
+ *
+ * This program is free software; you can redistribute it
+ * and/or modify it under the terms of the GNU General Public
+ * License (version 2) as published by the FSF - Free Software
+ * Foundation.
+ */
+#include "wm_agent_upgrade_tasks.h"
+#include "os_net/os_net.h"
+#include "shared.h"
+
+/* Hash table of current tasks based on agent_id */
+static OSHash *task_table_by_agent_id;
+
+wm_upgrade_task* wm_agent_init_upgrade_task() {
+    wm_upgrade_task *task;
+    os_malloc(sizeof(wm_upgrade_task), task);
+    task->custom_file_path = NULL;
+    task->custom_installer = NULL;
+    task->custom_version = NULL;
+    task->wpk_repository = NULL;
+    task->force_upgrade = false;
+    task->use_http = false;
+    task->state = NOT_STARTED;
+    return task;
+}
+
+void wm_agent_free_upgrade_task(wm_upgrade_task* task) {
+    os_free(task->custom_file_path);
+    os_free(task->custom_installer);
+    os_free(task->custom_version);
+    os_free(task->wpk_repository);
+    os_free(task);
+}
+
+void wm_agent_init_task_map() {
+    task_table_by_agent_id = OSHash_Create();
+}
+
+
+void wm_agent_destroy_task_map() {
+    OSHash_Free(task_table_by_agent_id);
+}
+
+void wm_agent_insert_tasks_id(const int task_id, const int agent_id) {
+    char agent_id_string[128];
+    sprintf(agent_id_string, "%d", agent_id);
+    wm_task *agent_task = (wm_task *)OSHash_Get_ex(task_table_by_agent_id, agent_id_string);
+    if (agent_task) {
```

Comment:
Shouldn't return an error if the task doesn't exist?

Reply:
I am going to add an assert just in case, but that situation should never happen on current design.